### ISSUE TITLE
Types & Grammar - Chapter 5: Grammar -- Misuse of term "proper subset"

### ISSUE THREAD

**Issue by amphrony** [getify/You-Dont-Know-JS#1376]
Book 4, Types & Grammar > Chapter 5: Grammar > Contextual Rules > Labels
https://github.com/getify/You-Dont-Know-JS/blob/master/types%20%26%20grammar/ch5.md#labels

In the first paragraph about JSON in the section on labeled statements:
> It's a very common belief that JSON is a proper subset of JS, so a string of JSON (like {"a":42} -- notice the quotes around the property name as JSON requires!) is thought to be a valid JavaScript program. Not true! Try putting {"a":42} into your JS console, and you'll get an error.

I think the meaning of the term "proper subset" is opposite its intended use here.

Per wolfram mathworld (http://mathworld.wolfram.com/ProperSubset.html):
> A proper subset of a set S [...] is a subset that is strictly contained in S and so necessarily excludes at least one member of S.

So JSON is, in fact, a proper subset of the JS language, since it is not valid JS grammar by itself.

This paragraph could instead say the following:
> It's a very common belief that JSON is **not** a proper subset of JS, so [...]

However, it would be incorrect to call JSON an "improper subset" of JS. Per wolfram mathworld (http://mathworld.wolfram.com/ImproperSubset.html):
> The subset consisting of all elements of a given set is called an improper subset (Kamke 1950, p. 6).

In other words, for a given set, there is only one improper subset, which is that set itself.

**Comment by getify** [comment]
I think you're misinterpreting that definition by wolfram, or at least misapplying it as I've used it in the book. 

The exclusion of an item from the superset doesn't mean that the subset is "missing something" to make it a valid set. Indeed, all parts of a subset, including its whole self, must be present and valid in the superset.

Moreover, the cited definition is strictly mathematical, but my usage is programming languages. They're related concepts, but not the same.

A subset of a programming language would hold that anything that is valid in the subset is valid in the superset.

That is not true of JSON. There are valid JSON strings which are not valid JS strings: `{"a": 1}`
